nupic/algorithms/anomaly_likelihood.py

```diff
@@ -204,18 +201,22 @@ def anomalyProbability(self, value, anomalyScore, timestamp=None):
           windowSize=self._historicalScores.maxlen,
           learningPeriod=self._claLearningPeriod)
 
-        _, _, self._distribution = (
-          estimateAnomalyLikelihoods(
-            self._historicalScores,
-            skipRecords=numSkipRecords)
-          )
+        _, _, self._distribution = estimateAnomalyLikelihoods(
+          self._historicalScores,
+          skipRecords=numSkipRecords)
+
+      likelihoods, _, self._distribution = updateAnomalyLikelihoods(
+        [dataPoint],
+        self._distribution)
 
-      likelihoods, _, self._distribution = (
-        updateAnomalyLikelihoods([dataPoint],
-                                 self._distribution)
-      )
       likelihood = 1.0 - likelihoods[0]
 
+      # If anomaly score > 0.99 then we greedily update the statistics.
```

Comment:
This says why it is ok but not why we do it in the first place. I believe that it is to mitigate the impact of not updating the distribution at every iteration. It basically says let us only periodically update it but if we have a very high anomaly likelihood then we need it to be accurate so override the periodic update and do an update now.